src/Components/Components/src/Rendering/RenderModeResolver.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Rendering;
+
```

Comment:
Conceptually, `RenderModeResolver` is a DI service that converts a rendermode (arbitrary object implementing `IComponentRenderMode`) into some concrete behavior. Each platform can have its own subclass that determines which rendermodes are supported and what to do with them.

Reply:
Merged into renderer